### ISSUE TITLE
Telemetry Data Exception

### ISSUE THREAD

**Issue by gsigler** [Learnosity/learnosity-sdk-asp.net#32]
I am trying to generate a security object without providing a request object. 

From your documentation:
request
An **optional** associative array^ of data relevant to the API being used. This will be any data minus the security details that you would normally use to initialise an API.

However, if I try to create an init without a request object it throws the following exception. 

` var init = new Init("items", securityJson, _config.ConsumerSecret);`

```
NullReferenceException: Object reference not set to an instance of an object.
LearnositySDK.Request.Init.addTelemetryData()
```

**Comment by flakeparadigm** [comment]
Thanks for the report, @gsigler. We've added this into our backlog for our development team to work on, and we'll make sure to provide updates on the progress as they become available. If this is time sensitive for your project, you're welcome to file a Pull Request yourself to help speed up the process.

**Comment by flakeparadigm** [comment]
Hi @gsigler,
To make sure we address this issue properly, are you able to provide more details about the particular request/initialization you're trying to build that you need to sign without providing the `request` to the `Init` constructor?

While I agree the documentation and the actual behaviour of the SDK do not align in this case, Items API requires a few basic parameters specified in the `request` (generally `user_id`, `session_id`, and a set of Items to use). Additionally, the `request` is used as an input for generating the signature so it cannot be altered after generation.

**Comment by gsigler** [comment]
Ahh that was not clear from the documentation. I was expecting to be able to create an Init and then attach the request information later. But it makes sense in retrospect that you can't change the body of the request after you create an init. 

In that case there is probably nothing to be done. 
Thanks!

**Comment by flakeparadigm** [comment]
No worries. Yeah, currently there isn't a way to update the request after calling the constructor.

I'll be keeping this issue open while we work through fixing the exception that is thrown, though, as a signed request should still be generated at least.

In this particular case, if the signing had worked and the generated request was used, Items API would have provided more helpful information about the required fields within the request.

**Comment by flakeparadigm** [comment]
No worries. I'll be keeping this ticket open until the exception you reported is resolved, though, as it should still generate a signature without error.

In this particular case, if the exception had not been thrown, sending the request to Items API would have resulted in more useful error messages with information about any required fields that may be missing from the request.

**Comment by flakeparadigm** [comment]
Hey @gsigler,
We've merged a fix for the issue you reported and plan to release it to NuGet on Monday. Please let us know if you have any further questions. (However if it's urgent, it's probably best to contact our support because they generally have a quicker response time)

**Comment by flakeparadigm** [comment]
We have completed the release containing a fix for the issue you have reported - v0.9.1. It is now available on NuGet for download and use.
